app/scripts/metamask-controller.js

```diff
@@ -2507,8 +2507,8 @@ export default class MetamaskController extends EventEmitter {
 
       // TODO: Remove once the preferences controller has been replaced with the core monorepo implementation
       this.controllerMessenger.publish(
-        `${this.preferencesController.name}:stateChange`,
-        [state, []],
+        'PreferencesController:stateChange',
+        state,
```

Comment:
These would be the arguments expected by this `publish` call:

```suggestion
        state,
        [],
```

or alternatively,

```suggestion
        ...[state, []],
```

For reference, this is the signature of the `publish` method of `RestrictedControllerMessenger`:

```ts
publish<EventType extends Event['type'] & NamespacedName<Namespace>>(
    event: EventType,
    ...payload: ExtractEventPayload<Event, EventType>
  )
```